src/main/java/org/mule/runtime/dsl/api/ConfigResource.java

```diff
@@ -91,7 +95,13 @@ public ConfigResource(String resourceName, InputStream inputStream) {
 
   public InputStream getInputStream() throws IOException {
     if (inputStream == null && url != null) {
-      inputStream = url.openStream();
+      URLConnection urlConnection = url.openConnection();
```

Comment:
Move this duplicated code to an utility function in org.mule.runtime.core.api.util.IoUtils